pysteps/blending/steps.py

```diff
@@ -1722,26 +1723,29 @@ def calculate_weights_spn(correlations, cov):
             # Ensure the correlation of the model with itself is always 1.0
             for i, _ in enumerate(cov):
                 cov[i][i] = 1.0
-            # Make a numpy matrix out of the array
-            cov_matrix = np.asmatrix(cov)
-            # Get the inverse of the matrix
-            cov_matrix_inv = cov_matrix.getI()
-            # The component weights are the dot product between cov_matrix_inv
-            # and cor_vec
-            weights = cov_matrix_inv.dot(correlations)
+            # Use a numpy array instead of a matrix
+            cov_matrix = np.array(cov)
+            # Get the inverse of the matrix using scipy's inv function
+            cov_matrix_inv = inv(cov_matrix)
+            # The component weights are the dot product between cov_matrix_inv and cor_vec
+            weights = np.dot(cov_matrix_inv, correlations)
             weights = np.nan_to_num(
                 weights, copy=True, nan=10e-5, posinf=10e-5, neginf=10e-5
             )
             # If the dot product of the weights with the correlations is
             # larger than 1.0, we assign a weight of 0.0 to the noise (to make
             # it numerically stable)
-            if weights.dot(correlations) > 1.0:
```

Comment:
Would it not be more efficient to compute np.dot(weights, correlations) once and then do the check as you need this variable both in the if and else statement.

Reply:
good catch! see [0919ba2](https://github.com/pySTEPS/pysteps/pull/407/commits/0919ba26f2b4ea801eac52958a1de01be01e81ff)